django/db/models/aggregates.py

```diff
@@ -155,14 +155,16 @@ class Count(Aggregate):
     name = "Count"
     output_field = IntegerField()
     allow_distinct = True
-    empty_result_set_value = 0
 
-    def __init__(self, expression, filter=None, **extra):
+    def __init__(self, expression, filter=None, default=0, **extra):
+        if default != 0:
+            raise TypeError(f"{self.__class__.__name__} does not allow default.")
+
```

Comment:
I wouldn't change the current `default=None` and always pass `default=0` to the super call, e.g.
```diff
diff --git a/django/db/models/aggregates.py b/django/db/models/aggregates.py
index ed1c648b31..5ede490a71 100644
--- a/django/db/models/aggregates.py
+++ b/django/db/models/aggregates.py
@@ -163,15 +163,15 @@ class Count(Aggregate):
     output_field = IntegerField()
     allow_distinct = True
 
-    def __init__(self, expression, filter=None, default=0, **extra):
-        if default != 0:
+    def __init__(self, expression, filter=None, default=None, **extra):
+        if default is not None:
             raise TypeError(f"{self.__class__.__name__} does not allow default.")
 
         if expression == "*":
             expression = Star()
         if isinstance(expression, Star) and filter is not None:
             raise ValueError("Star cannot be used with filter. Please specify a field.")
-        super().__init__(expression, filter=filter, default=default, **extra)
+        super().__init__(expression, filter=filter, default=0, **extra)
 
 
 class Max(Aggregate):
```
the same for `RegrCount`.

Reply:
I did it for sake of people who are using older versions and used `default=0` and trying to upgrade to new version.
In this way, there is no need to change the code and `default=0` will work.
What do you think ?

Reply:
As far as I'm previously it was not possible to set `default` for `Count()`.